src/rebuild/scan.c

```diff
@@ -717,6 +734,71 @@ rebuild_container_scan_cb(daos_handle_t ih, vos_iter_entry_t *entry,
 	return rc;
 }
 
+static int
+reint_cont_destroy_cb(daos_handle_t ih, vos_iter_entry_t *entry,
+			  vos_iter_type_t type, vos_iter_param_t *iter_param,
+			  void *data, unsigned *acts) {
+	struct rebuild_scan_xarg	*xarg = data;
+	struct rebuild_scan_arg		*arg = xarg->arg;
+	struct rebuild_tgt_pool_tracker *rpt = arg->rpt;
+	struct ds_cont_child		*ds_cont_child;
+	int				rc;
+
+	/* resync DTXs' status firstly. */
+	if (uuid_compare(xarg->co_uuid, entry->ie_couuid) == 0) {
+		D_DEBUG(DB_REBUILD, DF_UUID" already scan\n",
+			DP_UUID(xarg->co_uuid));
+		return 0;
+	}
+
+	rc = dtx_resync(iter_param->ip_hdl, rpt->rt_pool_uuid, entry->ie_couuid,
+			rpt->rt_rebuild_ver, true);
+	if (rc) {
+		D_ERROR(DF_UUID" dtx resync failed: rc %d\n",
+			DP_UUID(rpt->rt_pool_uuid), rc);
+		return rc;
+	}
+
+	rc = ds_cont_child_lookup(rpt->rt_pool_uuid, entry->ie_couuid,
+			&ds_cont_child);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to lookup container child."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = vos_cont_close(ds_cont_child->sc_hdl);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to close container."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(out, rc);
+	}
+
+	rc = vos_cont_destroy(iter_param->ip_hdl, entry->ie_couuid);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to destroy container."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(open, rc);
+	}
+
+	rc = vos_cont_create(iter_param->ip_hdl, entry->ie_couuid);
+	if (rc != 0) {
+		D_ERROR(DF_UUID"Reintegration failed to create container."
+				" rc: %d", DP_UUID(entry->ie_couuid), rc);
+		D_GOTO(out, rc);
+	}
+open:
+	rc = vos_cont_open(iter_param->ip_hdl, entry->ie_couuid,
+			&ds_cont_child->sc_hdl);
```

Comment:
This container/pool creation/open probably should be in initiate phase.

Reply:
I suppose we can just use the ds_cont_local_open() to create and open container, like what rebuild does today.

Actually I don't think we should call vos_create/destroy/open/close in rebuild module directly, rebuild (and all other server modules) should access vos container through ds_cont_child instead.

Reply:
I mean these container suppose to be opened for rebuild initiate phase ? so it should be moved to migrate part.  But I can do it in the following patch.